tests/saitests/py3/sai_qos_tests.py

```diff
@@ -1024,21 +1025,22 @@ def runTest(self):
                 print(list(map(operator.sub, pg_cntrs, pg_cntrs_base)),
                       file=sys.stderr)
                 for i in range(0, PG_NUM):
-                    if i == pg:
-                        if i == 0 or i == 4:
-                            assert (pg_cntrs[pg] >=
-                                    pg_cntrs_base[pg] + len(dscps))
+                    # DNX/Chassis:
+                    # pg = 0 => Some extra packets with unmarked TC
+                    # pg = 4 => Extra packets for LACP/BGP packets
+                    # pg = 7 => packets from cpu to front panel ports
+                    if platform_asic and platform_asic == "broadcom-dnx":
+                        if i == pg:
+                            assert (pg_cntrs[pg] >= pg_cntrs_base[pg] + len(dscps))
+                        elif i in [0, 4, 7]:
+                            assert (pg_cntrs[i] >= pg_cntrs_base[i])
                         else:
-                            assert (pg_cntrs[pg] ==
-                                    pg_cntrs_base[pg] + len(dscps))
+                            assert (pg_cntrs[i] == pg_cntrs_base[i])
                     else:
-                        # LACP packets are mapped to queue0 and tcp syn packets for BGP to queue4
-                        # So for those queues the count could be more
-                        if i == 0 or i == 4:
-                            assert (pg_cntrs[i] >= pg_cntrs_base[i])
+                        if i == pg:
```

Comment:
@ansrajpu-git  Can you please keep non DNX check same ?

Reply:
@vmittal-msft , As per the discussion and notes taken, I have implemented the changes.
the non-chassis behavior is reverted back to how it was originally implemented PR #7079 

@[vmittal-msft](https://github.com/vmittal-msft) vmittal-msft [2 weeks ago](https://github.com/sonic-net/sonic-mgmt/pull/9859#discussion_r1501120751)
As discussed, lets try this logic -

i => 0 -> 7
pg => 0,3,4
dscps => 62,1,1

DNX/Chassis:
pg = 0 => Some extra packets with unmarked  
pg = 4 => Extra packets for LACP/BGP packets
pg = 7 => packets from cpu to front panel ports

**if non chassis:
if i == pg:
assert(pg_cntrs[pg] == pg_cntrs_base[pg] + len(dscps))
else:
assert(pg_cntrs[i] == pg_cntrs_base[i])**
else
i = {3}
assert(pg_cntrs[pg] == pg_cntrs_base[pg] + len(dscps))
i = {0,4,7}
assert(pg_cntrs[pg] >= pg_cntrs_base[pg] + len(dscps))
else {1,2,5,6}
assert(pg_cntrs[i] == pg_cntrs_base[i])


Please confirm if more changes are required..

Reply:
@ansrajpu-git i still see some difference from the login we discussed. Can you please check ?

Reply:
@vmittal-msft, updated the logic. Please check.